private/buf/buffetch/proto_file_ref.go

```diff
@@ -0,0 +1,70 @@
+// Copyright 2020-2021 Buf Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package buffetch
+
+import (
+	"path/filepath"
+
+	"github.com/bufbuild/buf/private/buf/buffetch/internal"
+	"github.com/bufbuild/buf/private/pkg/normalpath"
+)
+
+var _ ProtoFileRef = &protoFileRef{}
+
+type protoFileRef struct {
+	protoFileRef internal.ProtoFileRef
+	dirPath      string
+}
+
+func newProtoFileRef(internalProtoFileRef internal.ProtoFileRef) *protoFileRef {
+	return &protoFileRef{
+		protoFileRef: internalProtoFileRef,
+		dirPath:      internalProtoFileRef.Path(),
+	}
+}
+
+func (r *protoFileRef) PathForExternalPath(externalPath string) (string, error) {
+	if r.dirPath == "" {
+		return normalpath.NormalizeAndValidate(externalPath)
+	}
```

Comment:
Based on the constructor, `dirPath` will always be non-empty, so I don't know if this condition is reachable. Should we just rely on the `protoFileRef` field and drop the `dirPath` field altogether?